src/Pim/Bundle/CatalogBundle/tests/integration/ElasticSearch/IndexConfiguration/PimCatalogNumberIntegration.php

```diff
@@ -0,0 +1,302 @@
+
+    public function testEmptyOperator()
+    {
+        $query = [
+            'query' => [
+                'bool' => [
+                    'must_not' => [
+                        'exists' => [
+                            'field' => 'values.box_quantity-decimal.<all_locales>.<all_channels>',
+                        ],
+                    ],
+                ],
+            ],
+        ];
+
+        $productsFound = $this->getSearchQueryResults($query);
+
+        $this->assertProducts($productsFound, ['product_7']);
+    }
+
+    public function testNotEmptyOperator()
+    {
+        $query = [
+            'query' => [
+                'bool' => [
+                    'filter' => [
+                        'exists' => [
+                            'field' => 'values.box_quantity-decimal.<all_locales>.<all_channels>',
+                        ],
+                    ],
+                ],
+            ],
+        ];
+
+        $productsFound = $this->getSearchQueryResults($query);
+
+        $this->assertProducts(
+            $productsFound,
+            ['product_1', 'product_2', 'product_3', 'product_4', 'product_5', 'product_6']
+        );
+    }
+
+    public function testSortAscending()
+    {
+        $query = [
+            'query' => [
+                'match_all' => new \stdClass(),
+            ],
+            'sort'  => [
+                [
+                    'values.box_quantity-decimal.<all_locales>.<all_channels>' => [
+                        'order'   => 'asc',
+                        'missing' => '_first',
+                    ],
+                ],
+            ],
+        ];
+
+        $productsFound = $this->getSearchQueryResults($query);
+
+        $this->assertProducts(
+            $productsFound,
+            ['product_7', 'product_2', 'product_5', 'product_6', 'product_1', 'product_4', 'product_3']
+        );
+    }
+
+    public function testSortDescending()
+    {
+        $query = [
+            'query' => [
+                'match_all' => new \stdClass(),
+            ],
+            'sort'  => [
+                [
+                    'values.box_quantity-decimal.<all_locales>.<all_channels>' => [
+                        'order'   => 'desc',
+                        'missing' => '_last',
+                    ],
+                ],
+            ],
+        ];
+
+        $productsFound = $this->getSearchQueryResults($query);
+
+        $this->assertProducts(
+            $productsFound,
+            ['product_3', 'product_4', 'product_1', 'product_6', 'product_5', 'product_2', 'product_7']
+        );
+    }
+
+    /**
+     * This method indexes dummy products in elastic search.
+     */
+    protected function addProducts()
+    {
+        $products = [
+            [
+                'identifier' => 'product_1',
+                'values'     => [
+                    'box_quantity-decimal' => [
+                        '<all_locales>' => [
+                            '<all_channels>' => '10.0',
```

Comment:
is it on purpose we mix strings and number? I guess yes
are we supposed to normalise everything as strings? (real question)